DSharpPlus.Rest/DiscordRestClient.cs

```diff
@@ -1010,9 +1012,11 @@ public Task DeleteGuildRoleAsync(ulong guild_id, ulong role_id, string reason)
         /// <param name="hoist">Whether this role should be hoisted</param>
         /// <param name="mentionable">Whether this role should be mentionable</param>
         /// <param name="reason">Reason why this role was created</param>
+        /// <param name="icon">The icon to add to this role</param>
+        /// <param name="emoji">The emoji to add to this role. Must be unicode.</param>
         /// <returns></returns>
-        public Task<DiscordRole> CreateGuildRoleAsync(ulong guild_id, string name, Permissions? permissions, int? color, bool? hoist, bool? mentionable, string reason)
-            => this.ApiClient.CreateGuildRoleAsync(guild_id, name, permissions, color, hoist, mentionable, reason);
+        public Task<DiscordRole> CreateGuildRoleAsync(ulong guild_id, string name, Permissions? permissions, int? color, bool? hoist, bool? mentionable, string reason, Stream icon = null, DiscordEmoji emoji = null)
+            => this.ApiClient.CreateGuildRoleAsync(guild_id, name, permissions, color, hoist, mentionable, reason, icon , emoji?.Id is 0 ? emoji?.ToString() : emoji?.Id.ToString());
```

Comment:
Are u sure about it?

In the client the guild emote is transmitted as base64..
I don't know how the api would handle it

Reply:
it sends base64 becuase it takes the image of the emoji and not the emoji itself.

Reply:
Transmitting the guild emoji id = clearing of icon

Reply:
Uh, no? transmitting the id doesn't do anything as far as I'm aware; this change has been reverted anyway, as I've realized the client is stupid and the tooltips within it should not be trusted.

Reply:
I tried it right now ^^
It does

https://github.com/discord/discord-api-docs/issues/3879
https://discord.com/channels/81384788765712384/381887113391505410/892218487701897246